File: e2e/cypress-javascript/interactions/pages/CreateTaskForm.js
class CreateTaskForm {
  get titleInput() {
    return cy.get('[data-testid="create-task-title-input"]');
  }

  get descriptionInput() {
    return cy.get('#description');
  }

  get statusDropdown() {
    return cy.get('[data-testid="status-dropdown"]');
  }

  get priorityDropdown() {
    return cy.get('[data-testid="priority-dropdown"]');
  }

  get submitButton() {
    return cy.get('[data-testid="create-button"]');
  }

  statusOption(status) {
    return cy.get(`[data-testid="status-dropdown-option-${status}"]`);
  }

  priorityOption(priority) {
    return cy.get(`[data-testid="priority-dropdown-option-${priority}"]`);
  }
}

module.exports = { CreateTaskForm };

File: e2e/capybara-ruby/interactions/pages/create_task_form.rb
# frozen_string_literal: true

require_relative '../../support/capybara_helpers'

class CreateTaskForm
  include CapybaraHelpers
  def title_input
    find('[data-testid="create-task-title-input"]')
  end

  def description_input
    find('#description')
  end

  def status_dropdown
    find('[data-testid="status-dropdown"]')
  end

  def priority_dropdown
    find('[data-testid="priority-dropdown"]')
  end

  def submit_button
    find('[data-testid="create-button"]')
  end

  def status_option(status)
    find("[data-testid=\"status-dropdown-option-#{status}\"]", visible: :all)
  end

  def priority_option(priority)
    find("[data-testid=\"priority-dropdown-option-#{priority}\"]", visible: :all)
  end
end
